Doc/library/logging.handlers.rst

```diff
@@ -382,7 +382,8 @@ timed intervals.
    The system will save old log files by appending extensions to the filename.
    The extensions are date-and-time based, using the strftime format
    ``%Y-%m-%d_%H-%M-%S`` or a leading portion thereof, depending on the
-   rollover interval.
+   rollover interval. In the case that an existing file has the same name,
+   a numbered suffix will be added to the new file to prevent overwriting logs.
```

Comment:
I was looking at other issues recently - bpo-43344, bpo-44753 - so this solution might not work in those scenarios. Specifcally, in scenarios where the user wants to keep the suffix as ".log" or similar. After changes to address those issues, rollover logic expects rolled-over files to start with the base filename (minus extension) + ".", and contain `self.extMatch`.  I've also added comments on the issue itself.